pkg/controller/template/render.go

```diff
@@ -480,10 +470,10 @@ func etcdMetricCertCommand(cfg RenderConfig) (interface{}, error) {
 }
 
 func cloudProvider(cfg RenderConfig) (interface{}, error) {
-	switch cfg.Platform {
-	case platformAWS, platformAzure, platformOpenStack, platformVSphere:
-		return cfg.Platform, nil
-	case platformGCP:
+	switch cfg.Infra.Status.PlatformStatus.Type {
+	case configv1.AWSPlatformType, configv1.AzurePlatformType, configv1.OpenStackPlatformType, configv1.VSpherePlatformType:
+		return strings.ToLower(string(cfg.Infra.Status.PlatformStatus.Type)), nil
+	case configv1.GCPPlatformType:
 		return "gce", nil
```

Comment:
I don't thiink this is currently used anywere. Might make sense to just move it into the case before and have it return "gcp"

Reply:
I err, this is passed to the kubeletconfig as cloudProvider, so nvmd